src/lib.rs

```diff
@@ -76,15 +76,23 @@ pub mod tokio_adapter;
 ///
 /// As of the time of this writing, the opentelemetry crate exposes no link information
 /// so this struct does not send link information.
-#[derive(Derivative)]
-#[derivative(Debug)]
 pub struct StackDriverExporter {
-  #[derivative(Debug = "ignore")]
   tx: futures::channel::mpsc::Sender<Vec<Arc<SpanData>>>,
   pending_count: Arc<AtomicUsize>,
   maximum_shutdown_duration: Duration,
 }
 
+impl fmt::Debug for StackDriverExporter {
+  fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+    fmt
+      .debug_struct("StackDriverExporter")
+      .field("tx", &"(elided)")
+      .field("pending_count", &self.pending_count)
+      .field("maximum_shutdown_duration", &self.maximum_shutdown_duration)
```

Comment:
One concern I see with trading a `derive`d `Debug` implementation with this hand-rolled one that specifically accesses fields via `self` is that a `derive` will never accidentally miss a newly added field, which with this code is a distinct possibility. There's a solution, though: destructuring `self`.

Reply:
My suggested edit here:

```suggestion
    #[allow(clippy::unneeded_field_pattern)]
    let Self {
      maximum_shutdown_duration,
      pending_count,
      tx: _,
    } = self;
    fmt
      .debug_struct("StackDriverExporter")
      .field("tx", &"(elided)")
      .field("pending_count", pending_count)
      .field("maximum_shutdown_duration", maximum_shutdown_duration)
```